zst/stream.go

```diff
@@ -0,0 +1,57 @@
+package zst
+
+import (
+	"errors"
+	"io"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/pkg/storage"
+	"github.com/brimdata/zed/zcode"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zst/column"
+)
+
+var ErrBadTypeNumber = errors.New("bad type number in ZST root reassembly map")
+
+// Stream reads a columnar ZST object to generate a stream of zed.Values.
+// It also has methods to read metadata for test and debugging.
+type Stream struct {
```

Comment:
Nit: I'd just fold this into Reader since I don't see the benefit to keeping it separate.

Reply:
Ok, cleaned this all up.